packages/client/src/route/customRoutes.tsx

```diff
@@ -27,24 +27,33 @@ import { t } from 'i18next'
 import React, { Suspense } from 'react'
 import { Route, Routes, useLocation } from 'react-router-dom'
 
-import { CustomRoute } from '@etherealengine/client-core/src/common/services/RouterService'
 import { LoadingCircle } from '@etherealengine/client-core/src/components/LoadingCircle'
 
-const CustomRoutes = ({ customRoutes }: { customRoutes: CustomRoute[] }) => {
+const CustomRoutes = ({ customRoutes }) => {
   if (customRoutes === null) return <Routes></Routes>
 
   const location = useLocation()
   const { pathname } = location
 
-  const matchedRoutes = customRoutes.filter((customRoute) => customRoute.route.split('/')[1] === pathname.split('/')[1])
+  // Improve loading by only using matched route
+  const matchedRoutes = customRoutes.filter((r) => {
+    return r.route.split('/')[1], pathname.split('/')[1]
+  })
 
   return (
     <Suspense fallback={<LoadingCircle message={t('common:loader.loadingCustom')} />}>
       <Routes>
-        {matchedRoutes.map((customRoute, idx) => {
-          const { route, component, props: p } = customRoute
+        {matchedRoutes.map((route, i) => {
```

Comment:
To remove the ambiguity with `route` defined at line 47, this can be changed to `routeItem`.